### PR TITLE
Drop Pascal GPU support.

### LINKED ISSUE #14629: Remove support for Pascal.
RSN 34 (https://docs.rapids.ai/notices/rsn0034/) dropped support for Pascal GPUs in RAPIDS. We need to clean up the cudf repo to remove outdated code and docs, like the [README](https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/README.md?plain=1#L65) and [gpu_utils.py](https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/python/cudf/cudf/utils/gpu_utils.py#L73) that support Pascal GPUs.

This affects both Python and C++ code.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Closes #14629.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
@vuule There are a couple places where the C++ code has special handling for disabling ZSTD compression on Pascal. Can you weigh in on whether that is safe to remove, since 24.02 will not support Pascal GPUs?

https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/cpp/tests/io/comp/decomp_test.cpp#L216

https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/cpp/src/io/comp/nvcomp_adapter.cpp#L554

**Comment by vuule** [comment]
> @vuule There are a couple places where the C++ code has special handling for disabling ZSTD compression on Pascal. Can you weigh in on whether that is safe to remove, since 24.02 will not support Pascal GPUs?
> 
> https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/cpp/tests/io/comp/decomp_test.cpp#L216
> 
> https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/cpp/src/io/comp/nvcomp_adapter.cpp#L554

Should be safe to remove. I'd keep `compute_capability_major` in `feature_status_parameters` even though it won't be used, in case we need another workaround at a later point. We can remove the code you linked above.

**Review by karthikeyann — DISMISSED** [review]
README.md of 
python/cudf/
python/cudf_kafka/
python/custreamz/
python/dask_cudf/
needs update as well.

**Comment by bdice** [comment]
> README.md of python/cudf/ python/cudf_kafka/ python/custreamz/ python/dask_cudf/ needs update as well.

@karthikeyann Can you check this? I already updated `custreamz` in this PR. The `cudf`, `cudf_kafka`, and `dask_cudf` READMEs are all just symlinks to the root `README.md` so they don't need additional changes.

File: python/cudf/cudf/utils/gpu_utils.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
```
Comment (bdice) [inline python/cudf/cudf/utils/gpu_utils.py:1]:
```suggestion
# Copyright (c) 2020-2024, NVIDIA CORPORATION.
```

**Review by mroeschke — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]

**Review by vuule — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Comment by bdice** [comment]
@karthikeyann I'm going to go ahead and merge this since we've had enough reviews and I _think_ your comment is already addressed. If you see additional changes that are needed, please let me know and I'll follow up. 👍

**Pull request merged**